packages/sources/layer2-sequencer-health/src/adapter.ts

```diff
@@ -0,0 +1,81 @@
+import { ExecuteFactory } from '@chainlink/types'
+import { Logger, Requester, Validator } from '@chainlink/ea-bootstrap'
+import { ExecuteWithConfig, InputParameters } from '@chainlink/types'
+import { ExtendedConfig, Networks, makeConfig } from './config'
+import { requestBlockHeight, getSequencerHealth } from './network'
+
+export const makeNetworkStatusCheck = (network: Networks) => {
+  let lastSeenBlock: { block: number; timestamp: number } = {
+    block: 0,
+    timestamp: 0,
+  }
+
+  const _isPastBlock = (block: number) => lastSeenBlock.block === block
+  const _isStaleBlock = (block: number, delta: number): boolean => {
+    return _isPastBlock(block) && Date.now() - lastSeenBlock.timestamp >= delta
+  }
+  const _isValidBlock = (block: number) => lastSeenBlock.block <= block
+  const _updateLastSeenBlock = (block: number): void => {
+    lastSeenBlock = {
+      block,
+      timestamp: Date.now(),
+    }
+  }
+
+  return async (delta: number): Promise<boolean> => {
+    const block = await requestBlockHeight(network)
+    if (!_isValidBlock(block)) throw new Error('Block found is previous to last seen')
```

Comment:
Shouldn't this return false instead of erroring? Seems like we'd want to raise a flag.

Reply:
Any error should be considered as a `false` on the consumer